src/dirwatcher.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const fsPromises = require('fs/promises');
+const { EventEmitter } = require('events');
+
+class DirWatcher extends EventEmitter {
+  constructor() {
+    super();
+    this.directoryContents = {};
+    this.newDirectoryContents = {};
+    this.changedFiles = [];
+  }
+
+  async readDir(path) {
+    return fsPromises.readdir(path);
+  }
+
+  async readStats(path, fileNames) {
+    return fileNames.map(fileName => fsPromises.stat(`${path}/${fileName}`));
+  }
+
+  async saveState(path) {
+    // read dir file names
+    const fileNames = await this.readDir(path);
+    this.changedFiles = fileNames;
+    this.directoryContents > 0
```

Comment:
You can pass data through `emit` itself, and not by emitting just a string, so `dirWatcher.getChangedFiles();` will be unnesesary in the importer.